### PR TITLE
More fixes for the redis queue

### PR DESCRIPTION
**Pull request by flavio**
It's incredible, but the old code had the redis server address hard coded to localhost!

I've also added more debugging information when an error occurs.

### CODE REVIEW

File: libgearman-server/plugins/queue/redis/queue.cc [hunk 1]
```diff
@@ -118,13 +118,13 @@ gearmand_error_t Hiredis::initialize()
   int service_port= atoi(service.c_str());
   if ((_redis= redisConnect("127.0.0.1", service_port)) == NULL)
   {
-    return gearmand_gerror("Could not connect to redis server", GEARMAND_QUEUE_ERROR);
+    return gearmand_log_gerror(GEARMAN_DEFAULT_LOG_PARAM, GEARMAND_QUEUE_ERROR, "Could not connect to redis server: %s", _redis->errstr);
```
Comment (SpamapS) [inline libgearman-server/plugins/queue/redis/queue.cc]: Can you please make this a bit more vertical? It's very wide, which makes it a bit confusing to read even with a very wide editor.

File: libgearman-server/plugins/queue/redis/queue.cc [hunk 2]
```diff
@@ -255,7 +258,7 @@ static gearmand_error_t _hiredis_replay(gearman_server_st *server, void *context
   redisReply *reply= (redisReply*)redisCommand(queue->redis(), "KEYS %s*", GEARMAND_QUEUE_GEARMAND_DEFAULT_PREFIX);
   if (reply == NULL)
   {
-    return gearmand_gerror("Failed to call KEYS during QUEUE replay", GEARMAND_QUEUE_ERROR);
+    return gearmand_log_gerror(GEARMAN_DEFAULT_LOG_PARAM, GEARMAND_QUEUE_ERROR, "Failed to call KEYS during QUEUE replay: %s", queue->redis()->errstr);
```
Comment (SpamapS) [inline libgearman-server/plugins/queue/redis/queue.cc]: Same width issue here.

File: libgearman-server/plugins/queue/redis/queue.cc [hunk 3]
```diff
@@ -116,7 +116,7 @@ Hiredis::~Hiredis()
 gearmand_error_t Hiredis::initialize()
 {
   int service_port= atoi(service.c_str());
-  if ((_redis= redisConnect("127.0.0.1", service_port)) == NULL)
```
Comment (SpamapS) [inline libgearman-server/plugins/queue/redis/queue.cc]: Wow, just wow

**Comment by SpamapS** [comment]
Thanks Flavio. Keep up the good work. I'd like to have those minor changes done, and then I'll gladly merge this, and I think it's time for a release.

**Comment by flavio** [comment]
Fixed

File: libgearman-server/plugins/queue/redis/queue.cc (hunk 3 above)
Comment (BrianAker) [inline libgearman-server/plugins/queue/redis/queue.cc]: There is un-merged Mongo patch which is of a similar flavor (though it came with more tests (eventually)).

Keep in mind, the durable queue is dog slow. There was a gen2 version, which I am unsure of where the code is, that was based off the Baidu extensions to functions. It required workers which acted as agents for the queue.

**Pull request merged**